src/PKSim.Presentation/Presenters/PopulationAnalyses/PopulationAnalysisParameterFieldsPresenter.cs

```diff
@@ -74,25 +73,18 @@ public void AddCovariate(string covariate)
       public override void FieldSelected(IPopulationAnalysisField populationAnalysisField)
       {
          base.FieldSelected(populationAnalysisField);
-         var parameterField = populationAnalysisField as PopulationAnalysisParameterField;
-         if (parameterField != null)
+         switch (populationAnalysisField)
          {
-            ParameterFieldSelected(this, new ParameterFieldSelectedEventArgs(parameterFromField(parameterField), parameterField));
-            return;
-         }
-
-         var covariateField = populationAnalysisField as PopulationAnalysisCovariateField;
-         if (covariateField != null)
-         {
-            CovariateFieldSelected(this, new CovariateFieldSelectedEventArgs(covariateField.Covariate, covariateField));
-            return;
+            case PopulationAnalysisParameterField parameterField:
```

Comment:
This new switch by type syntax is so much nicer. Liskov says we don't need it but when you do, men it's better